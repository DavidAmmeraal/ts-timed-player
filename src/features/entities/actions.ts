/**
 * actions.ts
 * Action creators for entities actions.
 */
import Types from 'Types';
import { createAction } from 'typesafe-actions';

export const createEntityAction = createAction(
  'entities/CREATE_ENTITY',
  resolve => (entityType: string, props: Types.EntityProps) =>
    resolve({
      entityType,
      props,
    }),
);

export const updateEntityAction = createAction(
  'entities/UPDATE_ENTITY',
  resolve => (entityType: string, props: Types.EntityProps) =>
    resolve({
      entityType,
      props,
    }),
);

export const deleteEntityAction = createAction(
  'entities/DELETE_ENTITY',
  resolve => (entityType: string, id: string) =>
    resolve({
      entityType,
      id,
    }),
);
