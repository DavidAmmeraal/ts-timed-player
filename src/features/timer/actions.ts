/**
 * actions.ts
 * Action creators for timer.
 */
import { action } from 'typesafe-actions';

import { TOGGLE, TICK, SET_RATE, SET_TIME } from './constants';

export const toggle = () => action(TOGGLE);
export const tick = (time: number) => action(TICK, time);
export const setRate = (rate: number) => action(SET_RATE, rate);
/**
 * Creates an SET_TIME time action.
 * @param time time in microseconds
 */
export const setTime = (time: number) => action(SET_TIME, time);
