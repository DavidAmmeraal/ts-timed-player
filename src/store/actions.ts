import { createAsyncAction } from 'typesafe-actions';

import { FETCH_STAGE_REQUEST, FETCH_STAGE_SUCCESS, FETCH_STAGE_ERROR } from './constants';

export const fetchStage = createAsyncAction(
  FETCH_STAGE_REQUEST,
  FETCH_STAGE_SUCCESS,
  FETCH_STAGE_ERROR,
)<void, {}, Error>();
