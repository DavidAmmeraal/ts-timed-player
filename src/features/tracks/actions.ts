import { TRACK_ENTITY_NAME } from './constants';
import { entitiesActions } from 'features/entities';
import { ITrack } from './models';

export const createTrackAction = (track:ITrack) => entitiesActions.createEntityAction(TRACK_ENTITY_NAME, track);
export const updateTrackAction = (id:string, props:Partial<ITrack>) => entitiesActions.updateEntityAction(TRACK_ENTITY_NAME, id, props);
export const deleteTrackAction = (id:string) => entitiesActions.deleteEntityAction(TRACK_ENTITY_NAME, id);
