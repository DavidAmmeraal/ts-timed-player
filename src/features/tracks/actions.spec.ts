import * as actions from './actions';

describe('createTrackAction()', () => {
    it('should create a new CREATE action.', () => {
        const track = {
            id: '123',
            start: new Date(1),
            end: new Date(3)
        }
        const action = actions.createTrackAction(track);
        expect(action).toMatchSnapshot();
    })
});

describe('updateTrackAction()', () => {
    it('should create a new UPDATE action.', () => {
        const action = actions.updateTrackAction('123', { start: new Date(2)});
        expect(action).toMatchSnapshot();
    })
});

describe('deleteTrackAction()', () => {
    it('should create a new DELETE action.', () => {
        const action = actions.deleteTrackAction('123');
        expect(action).toMatchSnapshot();
    })
});