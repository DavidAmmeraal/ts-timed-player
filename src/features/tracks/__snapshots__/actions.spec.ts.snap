// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTrackAction() should create a new CREATE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entity": Object {
      "end": 1970-01-01T00:00:00.003Z,
      "id": "123",
      "start": 1970-01-01T00:00:00.001Z,
    },
    "entityType": "Track",
  },
  "type": "entities/CREATE",
}
`;

exports[`deleteTrackAction() should create a new DELETE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entityType": "Track",
    "id": "123",
  },
  "type": "entities/DELETE",
}
`;

exports[`updateTrackAction() should create a new UPDATE action. 1`] = `
Object {
  "meta": undefined,
  "payload": Object {
    "entityType": "Track",
    "id": "123",
    "props": Object {
      "start": 1970-01-01T00:00:00.002Z,
    },
  },
  "type": "entities/UPDATE",
}
`;
